packages/kit/src/utils/http.js

```diff
@@ -1,21 +1,22 @@
-/**
- * @param {Record<string, string | string[] | undefined>} headers
- * @param {string} key
- * @returns {string | undefined}
- * @throws {Error}
- */
-export function get_single_valued_header(headers, key) {
-	const value = headers[key];
-	if (Array.isArray(value)) {
-		if (value.length === 0) {
-			return undefined;
+/** @param {Partial<import('types/helper').ResponseHeaders> | undefined} object */
+export function to_headers(object) {
+	const headers = new Headers();
+
+	if (object) {
+		for (const key in object) {
+			const value = object[key];
+
+			if (value) {
```

Comment:
personal style preference - I generally try invert like `if (!value) continue;` to avoid too many indents. (you could do the same above with `if (object)`

Reply:
it doesn't make sense to return early if `!object` because then you'd be duplicating `return headers`. but i added the `continue`